README.md

```diff
@@ -1 +1,108 @@
+If only one out of remote or local is updated out of these 3, the final revision will have all the unchanged data in it along with the update data from the update entity(Either remote or local). If more than one entity is updated on the same line, it'd throw a `ConflictException`.
+
+
+## Install
+
+The library can be installed via [composer](http://getcomposer.org).
+
+````JSON
+{
+  "name": "relaxedws/merge",
+  "description": "Library used to perform merges between normalized array structures.",
+  "require-dev": {
+    "phpunit/phpunit": "~4.0"
+  },
+  "autoload": {
+    "psr-4": {
+      "Relaxed\\Merge\\": "src/"
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "Relaxed\\Merge\\Test\\": "tests/"
+    }
+  }
+}
+````
+
+## Example
+
+After [installation](#install), we can perform a merge the following way:
+
+````php
+<?php
+
+namespace testing;
+
+require __DIR__ ."/vendor/autoload.php";
+
+use Relaxed\Merge\ThreeWayMerge;
+
+$original = [
+    'keyA' => [
+        0 => [
+            'keyB' => 'This is honey
+            like this',
+            'keyC' => 'This is however, not apple',
+        ],
+        1 => [
+            'keyB' => 'This is milk',
+            'keyC' => 'This is mango',
+        ],
+        2 => 'a little sugar',
+    ]
+];
+$local = [
+    'keyA' => [
+        0 => [
+            'keyB' => 'This is honeybb
+            like ti',
+            'keyC' => 'This is however, not apple',
+        ],
+        1 => [
+            'keyB' => 'This is milky milky',
+            'keyC' => 'This is mango',
+        ],
+        2 => 'a little coffee'
+    ]
+];
+$remote = [
+    'keyA' => [
+        0 => [
+            'keyB' => 'This is honey
+            like this',
+            'keyC' => 'This is however, not apple',
+        ],
+        1 => [
+            'keyB' => 'This is milk',
+            'keyC' => 'This is changed because of remote',
+        ],
+        2 => 'a little sugar',
+    ]
+];
+
+$multiline = new ThreeWayMerge();
+$new_arr = $multiline->performMerge($original, $local, $remote);
+````
+
+## Contributing
+
+We welcome all developers to come forward for use/test of this library.
+Though it's working well for almost all possible scenarios but we all know there are bugs
+in every code. Although we keep updating and testing the library regulary, but still if
+you encounter any bugs, please feel free to drop us a comment with as much explaination as
+possible to reproduce the errors or you can even patch that.
+
+Other than directly working with the code, you can also help us with better documentation, fixing typo errors
+or adding more content to the `readme` file.
+
+We are open to feedback, suggestions and questions.
+We are always present at ``#drupal8-ports` on `irc.freenode.net` .
```

Comment:
Let's not mix in Drupal 8 here. If people wants to contribute to this library we should encourrage them to submit pull requests. And if they have questions they should file Github issues for this project. That's better, because then other people can search among these issues and maybe find answers to their own questions etc :)